python/cudf/cudf/_lib/pylibcudf/column.pyx

```diff
@@ -223,6 +223,42 @@ cdef class Column:
             c_result = move(make_column_from_scalar(dereference(c_scalar), size))
         return Column.from_libcudf(move(c_result))
 
+    @staticmethod
+    def from_cuda_array_interface_obj(object obj):
+        """Create a Column from an object with a CUDA array interface.
+
+        Parameters
+        ----------
+        obj : object
+            The object with the CUDA array interface to create a column from.
+
+        Returns
+        -------
+        Column
+            A Column containing the data from the CUDA array interface.
+
+        Note that data is not copied when creating the column. The caller is
+        responsible for ensuring the data is not mutated unexpectedly while the
+        column is in use.
+
+        """
+        if not hasattr(obj, '__cuda_array_interface__'):
+            raise ValueError("Object does not have a CUDA array interface")
+
+        data = gpumemoryview(obj)
+        iface = data.__cuda_array_interface__()
+        data_type = _data_type_from_iface(iface)
+        size = iface['shape'][0]
+        return Column(
+            data_type,
+            size,
+            data,
+            None,
```

Comment:
There's no way for a `__cuda_array_interface__` to advertise a mask is there?

Reply:
I used the [numba docs](https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html) as a reference which describes an optional `mask`. It refers out to [the array interface](https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.interface.html#__array_interface__). I expect we're one of the only libraries that would consider masked device arrays at this time, so it's probably not something we need to support in the short term. But the code should probably error just in case.

Reply:
Could the mask be passed here anyways even if it's not supported? Nice to have one less place to update when mask is supported

Reply:
The pylibcudf column needs the mask as a bitmask, but I think the array interface spec is not helpful in defining what the mask is:

> None or an object exposing the array interface. All elements of the mask array should be interpreted only as true or not true indicating which elements of this array are valid. The shape of this object should be “broadcastable” to the shape of the original array

So it is valid to pass an integer array `[0, 1, 5, 6, 2]` as the mask by my reading.

So the minimal thing to do here is to `raise` if `mask is not None`.